src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -660,6 +660,12 @@ private ObjectSpec CreateObjectSpec(TypeParseInfo typeParseInfo)
                         if (member is IPropertySymbol { IsIndexer: false, IsImplicitlyDeclared: false } property && !IsUnsupportedType(property.Type))
                         {
                             string propertyName = property.Name;
+
+                            if (properties?.ContainsKey(propertyName) is true)
```

Comment:
While using the most derived member fixes the issue at hand, I have found in my experiments that it can introduce regressions in virtual properties where the most derived member only overrides part of the base signature, for example:

```C#
class Base
{
    public virtual int X { get; set; }
}

class Derived1 : Base
{
    public override int X => 42;
}

class Derived2 : Base
{
    public override int X { set { base.X = value + 1; } }
}

```

In this case, the most derived IPropertySymbol for `Derived1` will report that `X` does not have a setter, which should result in the CB source generator ignoring it.

When the source generator checks if the particular property has an accessible setter (it doesn't appear to happen here, but I'm sure it does happen somewhere), this should be done against the base property definition (obtainable by recursively traversing the `IPropertySymbol.OverriddenProperty` property) which reports the actual property signature. Here's an [example of this](https://github.com/eiriktsarpalis/typeshape-csharp/blob/f6849e3fb7273bcbddd5405aeb42cc0d83fb072f/src/TypeShape.Roslyn/ModelGenerator/TypeDataModelGenerator.Object.cs#L119-L131) from my prototypes.